auto_submit/lib/service/validation_service.dart

```diff
@@ -146,9 +146,26 @@ class ValidationService {
       }
     }
     // If we got to this point it means we are ready to submit the PR.
-    bool processed = await processMerge(config, result, messagePullRequest);
-    if (processed) await pubsub.acknowledge('auto-submit-queue-sub', ackId);
-    log.info('Ack the processed message : $ackId.');
+    try {
+      bool processed = await processMerge(config, result, messagePullRequest);
+      if (processed) {
+        await pubsub.acknowledge('auto-submit-queue-sub', ackId);
+        log.info('Ack the processed message : $ackId.');
+      } else {
+        String message = 'Unable to merge pull request $prNumber.';
+        log.warning(message);
+        await gitHubService.removeLabel(slug, prNumber, Config.kAutosubmitLabel);
+        await gitHubService.createComment(slug, prNumber, message);
```

Comment:
Nit: encapsulate these repeated logics in a sub-function.

Reply:
Done.